### ISSUE TITLE
System tray icon is unresponsive in Linux

### ISSUE THREAD

**Issue by rhanneken** [keybase/client#25823]
When I click on the system tray icon, nothing happens. It doesn't matter if I left-click, right-click, or double-click. If the desktop client window is closed and I want to open it, I have to restart the app.

I'm using Keybase 6.2.2 on Pop!_OS 22.04 LTS (a Linux distribution downstream from Ubuntu 22.04 LTS). My desktop environment is GNOME 42.9.

This issue is similar to issue #23150, but in that case users were reporting that double-clicking worked, which it does not in my case.

**Comment by jravetch** [comment]
I have the exact same environment as @rhanneken and experiencing the same issue that the tray icon no longer responds.

**Comment by Espionage724** [comment]
I'm assuming it's a GNOME-specific thing and likely related to AppIndicator and KStatusNotifierItem tray icons being deprecated.

For an unrelated app (Jagex Launcher) if I enable the legacy AppIndicator and KStatusNotifierItem support via GNOME extensions on Fedora, I can get it's tray icon to show in the top-right area, but it doesn't interact at all with left/right click. If I disable the legacy tray support, the launcher ends up docking the icon to a separate floating Wine tray and works fine. Iirc Steam from RPM Fusion installs the legacy tray support as a dependency.

I'm not sure how Pop!_OS handles that. Could try toggling between X and Wayland if it offers that.

**Comment by rhanneken** [comment]
@Espionage724 The package `gnome-shell-extension-appindicator` is installed on my machine. The project page [says](https://github.com/ubuntu/gnome-shell-extension-appindicator):

>This extension integrates Ubuntu AppIndicators and KStatusNotifierItems (KDE's blessed successor of the systray) into GNOME Shell. Including support for legacy tray icons.

So I think that's not the issue.

Are you running Keybase on Linux? Does the system tray icon respond to your clicks? What's your desktop environment?

**Comment by Espionage724** [comment]
Right-clicking Keybase's icon in the top-right bar doesn't do anything, but double-clicking with left shows a small Keybase window with chats and files.

I'm using Fedora 38 with GNOME 44.3 and `gnome-shell-extension-appindicator` enabled.

**Comment by rhanneken** [comment]
@Espionage724 What version of Keybase are you running?

**Comment by Espionage724** [comment]
`keybase-6.2.2.20230726175256.4464bfb32d-1.x86_64`

**Comment by joakime** [comment]
I have the same issue, and the tray icon is totally unresponsive (single click, left/right/middle, double-click, doesn't matter, nothing happens).

Running: standard Ubuntu 22.04.3
Keybase: 6.2.2-20230726175256+4464bfb32d

Are there any logs somewhere I can gather information from to help the developers here?

**Comment by jravetch** [comment]
Updated to `6.2.3`, same issue

**Comment by 238Ver** [comment]
The same issue

**Comment by enrique-argones** [comment]
Same here.

**Comment by TestardR** [comment]
Same issue on Keybase `6.2.4-20240101011938+ae7e4a1c15`

**Comment by longsleep** [comment]
Well this now became a problem since one cannot simply continue using 6.0.x (where this works and was my solution so far).

Reason is that old versions contain expired embedded certificate (https://github.com/keybase/keybase-issues/issues/4219) so now a fix or workaround is really needed to be able to continue using Keybase.

**Comment by framerate** [comment]
Looking at the date of this issue, I have a bad feeling about the future of keybase

**Comment by Espionage724** [comment]
> Looking at the date of this issue, I have a bad feeling about the future of keybase

Tray icons were deprecated on GNOME years ago afaik and things still doing it today use a 3rd-party GNOME extension for the legacy tray support.

If Keybase's tray icon works fine on another DE like Xfce, then I would put the blame at GNOME for no official tray support, and then just let it go because Keybase works fine without a tray icon.

**Comment by rhanneken** [comment]
Pop!_OS 24.04 LTS was finally released. I upgraded and … the system tray icon is still unresponsive. ☹️ Note that the desktop environment is now COSMIC instead of GNOME.

**Comment by rhanneken** [comment]
Today I upgraded to Keybase 6.5.1-20260204010455.5c615aa1ce, and now the system tray icon is responsive! Clicking on it opens the mini-Keybase view.

Shall I close this issue? Is the problem solved for everyone else who reported it?
